include/soci/exchange-traits.h

```diff
@@ -44,12 +44,15 @@ struct exchange_traits
     };
 };
 
+#if SOCI_OS != SOCI_OS_SUN
+// As in SunOS: typedef char int8_t;
```

Comment:
I don't think it's a property of Solaris per se, we should almost certainly check whether `char` and `int8_t` are the same in CMake and define some `SOCI_HAVE_INT8_T_TYPEDEF` or something like this and test for it here.